<a href="https://colab.research.google.com/github/jnrxvr/python-curso-em-video/blob/main/Python_Mundo2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# NOTEBOOK DOS EXERCÍCIOS DO MUNDO 2 DE PYTHON DO CURSO EM VÍDEO

## Exercício Python 36: Escreva um programa para aprovar o empréstimo bancário para a compra de uma casa. Pergunte o valor da casa, o salário do comprador e em quantos anos ele vai pagar. A prestação mensal não pode exceder 30% do salário ou então o empréstimo será negado.

In [ ]:
valor_da_casa = float(input('Digite o valor da casa: '))
salario = float(input('Digite o salário: '))
anos = float(input('Digite o total de anos a pagar: '))

parcelas = valor_da_casa / (anos * 12)

if parcelas <= (salario * 0.30):
  print('Parabéns! Seu empréstimo foi aprovado!')
else:
  print('Lamento! Seu empréstimo não foi aprovado dessa vez. Tente novamente no futuro!')

Digite o valor da casa: 60000
Digite o salário: 5000
Digite o total de anos a pagar: 3
Lamento! Seu empréstimo não foi aprovado dessa vez. Tente novamente no futuro!


## Exercício Python 37: Escreva um programa em Python que leia um número inteiro qualquer e peça para o usuário escolher qual será a base de conversão: 1 para binário, 2 para octal e 3 para hexadecimal.

In [ ]:
num = int(input('Digite um número inteiro: '))

print(bin(num))
print(oct(num))
print(hex(num))

Digite um número inteiro: 10
0b1010
0o12
0xa


##Exercício Python 038: Escreva um programa que leia dois números inteiros e compare-os. mostrando na tela uma mensagem:

###– O primeiro valor é maior
###– O segundo valor é maior
###– Não existe valor maior, os dois são iguais

In [ ]:
num1 = int(input('Digite o primeiro número inteiro: '))
num2 = int(input('Digite o segundo número inteiro: '))

if num1 > num2:
  print(f'O primeiro número {num1} é o maior.')
elif num2 > num1:
  print(f'O segundo número {num2} é o maior.')
else:
  print('Não existe valor maior, ambos são iguais.')

Digite o primeiro número inteiro: 10
Digite o segundo número inteiro: 10
Não existe valor maior, ambos são iguais.


## Exercício Python 39: Faça um programa que leia o ano de nascimento de um jovem e informe, de acordo com a sua idade, se ele ainda vai se alistar ao serviço militar, se é a hora exata de se alistar ou se já passou do tempo do alistamento. Seu programa também deverá mostrar o tempo que falta ou que passou do prazo.

In [ ]:
from datetime import date
anoAtual = date.today().year # para se extrair o ano da data autal, não se utiliza os parênteses ao final do year
anoNascimento = int(input('Informe o seu ano de nascimento: '))

idade = anoAtual - anoNascimento
dif = 18 - idade
if idade < 18:
  if dif <= 1:
    print(f'\nVocê tem {idade} anos. Falta {dif} ano para você se alistar!')
  else:
    print(f'\nVocê tem {idade} anos. Faltam {dif} anos para você se alistar!')
elif idade == 18:
  print(f'Você tem {idade} anos. Está na hora de se alistar!')
else:
  if abs(dif) == 1:
    print(f'\nVocê tem {idade} anos. Já passou {abs(dif)} ano da idade do seu alistamento!')
  else:
    print(f'\nVocê tem {idade} anos. Já passaram {abs(dif)} anos da idade do seu alistamento!')


Informe o seu ano de nascimento: 2008

Você tem 16 anos. Faltam 2 anos para você se alistar!


## Exercício Python 040: Crie um programa que leia duas notas de um aluno e calcule sua média, mostrando uma mensagem no final, de acordo com a média atingida:

###– Média abaixo de 5.0: REPROVADO
###– Média entre 5.0 e 6.9: RECUPERAÇÃO
###– Média 7.0 ou superior: APROVADO

In [ ]:
nota1 = float(input('Digite a primeira nota do(a) aluno(a): '))
nota2 = float(input('Digite a segunda nota do(a) aluno(a): '))

media = (nota1 + nota2)/2

if media >= 7:
  print(f'A média do(a) aluno(a) foi {media} e ele(a) está APROVADO(A)!')
elif media >= 5 and media < 7:
  print(f'A média do(a) aluno(a) foi {media} e ele(a) está de RECUPERAÇÃO!')
else:
  print(f'A média do(a) aluno(a) foi {media} e ele(a) está REPROVADO(A)!')

KeyboardInterrupt: Interrupted by user

##Exercício Python 041: A Confederação Nacional de Natação precisa de um programa que leia o ano de nascimento de um atleta e mostre sua categoria, de acordo com a idade:

###– Até 9 anos: MIRIM
###– Até 14 anos: INFANTIL
###– Até 19 anos: JÚNIOR
###– Até 25 anos: SÊNIOR
###– Acima de 25 anos: MASTER

In [ ]:
idade = int(input('Digite a idade do(a) atleta: '))

if idade <= 9:
  print('CATEGORIA: MIRIM')
elif idade <= 14:
  print('CATEGORIA: INFANTIL')
elif idade <= 19:
  print('CATEGORIA: JÚNIOR')
elif idade <= 25:
  print('CATEGORIA: SÊNIOR')
else:
  print('CATEGORIA: MASTER')

Digite a idade do(a) atleta: 56
CATEGORIA: MASTER


## Exercício Python 42: Refaça o DESAFIO 35 dos triângulos, acrescentando o recurso de mostrar que tipo de triângulo será formado:

### – EQUILÁTERO: todos os lados iguais
### – ISÓSCELES: dois lados iguais, um diferente
### – ESCALENO: todos os lados diferentes

In [ ]:
r1 = float(input('Digite o comprimento da primeira reta: '))
r2 = float(input('Digite o comprimento da segunda reta: '))
r3 = float(input('Digite o comprimento da terceira reta: '))

if (r1 <(r2 + r3)) and (r2 < (r1 + r3)) and (r3 < (r1 + r2)): # testando se as três retas podem formar um triângulo
  print('É possivel formar um triângulo com esses segmentos de reta!')
  if r1 == r2 and r1 == r3: # verificando o tipo de triângulo de acordo com seus lados
    print('Esses comprimentos de reta formam um triângulo equilátero!')
  elif r1 == r2 and r1 != r3:
    print('Esses comprimentos de reta formam um triângulo isósceles!')
  else:
    print('Esses comprimentos de reta formam um triângulo escaleno!')
else:
  print('NÃO é possivel formar um triângulo com esses segmentos de reta!')

Digite o comprimento da primeira reta: 6
Digite o comprimento da segunda reta: 7
Digite o comprimento da terceira reta: 8
É possivel formar um triângulo com esses segmentos de reta!
Esses comprimentos de reta formam um triângulo escaleno!


## Exercício Python 43: Desenvolva uma lógica que leia o peso e a altura de uma pessoa, calcule seu Índice de Massa Corporal (IMC) e mostre seu status, de acordo com a tabela abaixo:

### – IMC abaixo de 18,5: Abaixo do Peso
### – Entre 18,5 e 25: Peso Ideal
### – 25 até 30: Sobrepeso
### – 30 até 40: Obesidade
### – Acima de 40: Obesidade Mórbida

In [ ]:
from math import pow

peso = float(input('Digite o peso da pessoa (em kilos): '))
altura = float(input('Digite a altura da pessoa (em metros): '))

imc = peso/(pow(altura, 2))

if imc < 18.5: # utilizando o arredondamento {imc:.2f} o resultado diverge do resultado correto, pois ele arrendonda números próximos a 18.5 como 18.50 e acaba considerando 18.50 como abaixo do peso
  print(f'IMC {imc}: abaixo do peso')
elif imc <= 25:
  print(f'IMC {imc}: peso ideal')
elif imc <= 30:
  print(f'IMC {imc}: sobrepeso')
elif imc <= 40:
  print(f'IMC {imc}: obesidade')
else:
  print(f'IMC {imc}: obesidade mórbida')

Digite o peso da pessoa (em kilos): 56.65
Digite a altura da pessoa (em metros): 1.75
IMC 18.50: abaixo do peso


## Exercício Python 44: Elabore um programa que calcule o valor a ser pago por um produto, considerando o seu preço normal e condição de pagamento:

### – à vista dinheiro/cheque: 10% de desconto
### – à vista no cartão: 5% de desconto
### – em até 2x no cartão: preço formal
### – 3x ou mais no cartão: 20% de juros

In [ ]:
preco = float(input('Informe o valor do produto: '))
formaPagamento = int(input('Informe a condição de pagamento: '
                            '\n [1] - À vista dinheiro ou pix (10% de desconto)'
                            '\n [2] - À vista no cartão (5% de desconto)'
                            '\n [3] - Até 2x s/ juros no cartão'
                            '\n [4] - 3x ou mais no cartão (20% de juros)'
                            '\n Opção desejada: '))
print('*'*50)

if formaPagamento == 1:
  print(f'Opção escolhida: {formaPagamento}.')
  print('À vista dinheiro ou pix (10% de desconto): \n')
  print(f'Valor final com 10% de desconto: R$ {preco*0.9}.')
elif formaPagamento == 2:
  print(f'Opção escolhida: {formaPagamento}.')
  print('À vista no cartão (5% de desconto): \n')
  print(f'Valor final com 5% de desconto: R$ {preco*0.95}.')
elif formaPagamento == 3:
  print(f'Opção escolhida: {formaPagamento}.')
  print('Até 2x s/ juros no cartão: \n')
  print(f'Valor final: R$ {preco}.')
elif formaPagamento == 4:
  print(f'Opção escolhida: {formaPagamento}.')
  print('3x ou mais no cartão (20% de juros): \n')
  print(f'Valor final com juros de 20%: R$ {preco*1.2}.')
else:
  print('Opção inválida! Tente novamente.')

Informe o valor do produto: 100
Informe a condição de pagamento: 
 [1] - À vista dinheiro ou pix (10% de desconto)
 [2] - À vista no cartão (5% de desconto)
 [3] - Até 2x s/ juros no cartão
 [4] - 3x ou mais no cartão (20% de juros)
 Opção desejada: 4
**************************************************
Opção escolhida: 4.
3x ou mais no cartão (20% de juros): 

Valor final com juros de 20%: R$ 120.0.


## Exercício Python 45: Crie um programa que faça o computador jogar "Pedra, Papel e Tesoura" com você

In [ ]:
from random import choice

humano = input('Digite Pedra, Papel ou Tesoura: ').strip().upper()
computador = ['PEDRA', 'PAPEL', 'TESOURA']
escolhaPC = choice(computador)

if humano == 'PEDRA':
  if escolhaPC == 'TESOURA':
    print(f'Escolha humano: {humano}.')
    print(f'Minha escolha: {escolhaPC}.')
    print(f'VOCÊ VENCEU! PARABÉNS!')
  elif escolhaPC == 'PAPEL':
    print(f'Escolha humano: {humano}.')
    print(f'Minha escolha: {escolhaPC}.')
    print(f'MUAHAHAHA EU VENCI!')
  else:
    print(f'Escolha humano: {humano}.')
    print(f'Minha escolha: {escolhaPC}.')
    print(f'EMPATAMOS!')
elif humano == 'TESOURA':
  if escolhaPC == 'PAPEL':
    print(f'Escolha humano: {humano}.')
    print(f'Minha escolha: {escolhaPC}.')
    print(f'VOCÊ VENCEU! PARABÉNS!')
  elif escolhaPC == 'PEDRA':
    print(f'Escolha humano: {humano}.')
    print(f'Minha escolha: {escolhaPC}.')
    print(f'MUAHAHAHA EU VENCI!')
  else:
    print(f'Escolha humano: {humano}.')
    print(f'Minha escolha: {escolhaPC}.')
    print(f'EMPATAMOS!')
else:
  if escolhaPC == 'PEDRA':
    print(f'Escolha humano: {humano}.')
    print(f'Minha escolha: {escolhaPC}.')
    print(f'VOCÊ VENCEU! PARABÉNS!')
  elif escolhaPC == 'TESOURA':
    print(f'Escolha humano: {humano}.')
    print(f'Minha escolha: {escolhaPC}.')
    print(f'MUAHAHAHA EU VENCI!')
  else:
    print(f'Escolha humano: {humano}.')
    print(f'Minha escolha: {escolhaPC}.')
    print(f'EMPATAMOS!')

KeyboardInterrupt: Interrupted by user

## Exercício Python 46: Faça um programa que mostre na tela uma contagem regressiva para o estouro de fogos de artifício, indo de 10 até 0, com uma pausa de 1 segundo entre eles.

In [ ]:
# aula 13
import time

for c in range(10, -1, -1):
  print(c)
  time.sleep(1)
print('FELIZ ANO NOVO!')

10
9
8
7
6
5
4
3
2
1
0
FELIZ ANO NOVO!


## Exercício Python 47: Crie um programa que mostre na tela todos os números pares que estão no intervalo entre 1 e 50.

In [ ]:
for num in range(2, 51, 2):
  print(num)

2
4
6
8
10
12
14
16
18
20
22
24
26
28
30
32
34
36
38
40
42
44
46
48
50


## Exercício Python 48: Faça um programa que calcule a soma entre todos os números que são múltiplos de três e que se encontram no intervalo de 1 até 500.

In [ ]:
soma = 0
for num in range(3, 500, 3):
    # print(num) - apenas para verificar se estava exibindo os valores certos
    soma = soma + num
print(soma)

41583


Exercício Python 48.1: Faça um programa que calcule a soma entre todos os números ímpares que são múltiplos de três e que se encontram no intervalo de 1 até 500.

In [ ]:
soma = 0
for num in range(1, 501, 2): # começa em 1 então basta pular de 2 em 2 para ser ímpar
    if num % 3 == 0: # serve para chegar o número é múltiplo de 3, para isso basta dividí-lo por 3
     # print(num) - apenas para mostrar se o código estava calculando os valores corretos
      soma = soma + num
print(soma)

20667


## Exercício Python 49: Refaça o DESAFIO 9, mostrando a tabuada de um número que o usuário escolher, só que agora utilizando um laço for.

In [ ]:
num = int(input('Digite um número para saber sua tabuada: '))

for c in range(1, 11):
  print(f'{num} x {c} = {num*c}')

Digite um número para saber sua tabuada: 5
5 x 1 = 5
5 x 2 = 10
5 x 3 = 15
5 x 4 = 20
5 x 5 = 25
5 x 6 = 30
5 x 7 = 35
5 x 8 = 40
5 x 9 = 45
5 x 10 = 50


## Exercício Python 50: Desenvolva um programa que leia seis números inteiros e mostre a soma apenas daqueles que forem pares. Se o valor digitado for ímpar, desconsidere-o.

In [ ]:
soma = 0
for c in range(1, 7): # começa em 1 então basta pular de 2 em 2 para ser ímpar
      num = int(input(f'Digite o {c}º número: '))
      if num % 2 == 0:
        soma = soma + num
print(soma)

Digite o 1º número: 13
Digite o 2º número: 2
Digite o 3º número: 45
Digite o 4º número: 4536
Digite o 5º número: 54
Digite o 6º número: 5
4592


## Exercício Python 51: Desenvolva um programa que leia o primeiro termo e a razão de uma PA. No final, mostre os 10 primeiros termos dessa progressão.

In [ ]:
pTermo = int(input('Digite o primeiro termo da P.A.: '))
razao = int(input('Digite a razão da P.A.: '))

soma = pTermo
print(f'1º termo: {pTermo}')
for c in range(2, 11): # nesse caso, sabemos a quantidade máxima de termos
  soma = soma + razao
  print(f'{c}º termo: {soma}')

Exercício Python 51.1: Desenvolva um programa que leia o primeiro termo e a razão de uma PA. No final, mostre quantos termos o usuário desejar:

In [ ]:
pTermo = int(input('Digite o primeiro termo da P.A.: '))
razao = int(input('Digite a razão da P.A.: '))
qtdTermos = int(input('Digite a quantidade de termos da P.A.: '))

soma = pTermo
print(f'1º termo: {pTermo}')
for c in range(2, qtdTermos + 1):
  soma = soma + razao
  print(f'{c}º termo: {soma}')

## Exercício Python 52: Faça um programa que leia um número inteiro e mostre, de acordo com seus divisores, se ele é primo ou não!

In [ ]:
num = int(input('Digite um número: '))
for n in range(1, num+1):
  if num % n == 0:
    print(f'\33[33m', end='') # printará de amarelo se o número for divisor de núm
  else:
    print(f'\33[31m', end='') # printará de amarelo se o número NÃO for divisor de núm
  print(f'{n}', end=' ')

Digite um número: 12
1 2 3 4 5 6 7 8 9 10 11 12 

## Exercício Python 53: Crie um programa que leia uma frase qualquer e diga se ela é um palíndromo, desconsiderando os espaços. Exemplos de palíndromos:

#### APOS A SOPA, A SACADA DA CASA, A TORRE DA DERROTA, O LOBO AMA O BOLO, ANOTARAM A DATA DA MARATONA.

In [ ]:
# resolução da vídeo-aula

frase = input('Digite uma frase: ').strip().lower()
palavras = frase.split() # separando a frase em um vetor/lista através do separado espaço (' ')
junto = ''.join(palavras) # unindo todas as palavras em uma string só
inverso = ''
for letra in range((len(junto)-1), -1, -1): # para cada letra lida de trás para frente, a variável "inverso" receberá a 'letra" lida
  inverso += junto[letra]
if inverso == junto:
  print('Temos um palíndromo!')
else:
  print('A frase digitada não é um palíndromo!')

Digite uma frase: APOS A SOPA
Temos um palíndromo!


In [ ]:
# minha resolução inicial

frase = input('Digite uma frase: ').strip().lower()
fraseSemEspaco = frase.replace(' ', '')
print(fraseSemEspaco)
fraseInvertida = frase[::-1].replace(' ','').strip()
print(fraseInvertida)

if fraseSemEspaco == fraseInvertida:
  print('Temos um palíndromo!')
else:
  print('A frase digitada não é um palíndromo!')

Digite uma frase: O LOBO AMA O BOLO
oloboamaobolo
oloboamaobolo
Temos um palíndromo!


In [ ]:
# resolução mais simples da video-aula

frase = input('Digite uma frase: ').strip().lower()
fraseSemEspaco = frase.replace(' ', '')
fraseInvertida = fraseSemEspaco[::-1]

if fraseSemEspaco == fraseInvertida:
  print('Temos um palíndromo!')
else:
  print('A frase digitada não é um palíndromo!')

Digite uma frase: O LOBO AMA O BOLO
Temos um palíndromo!


## Exercício Python 54: Crie um programa que leia o ano de nascimento de sete pessoas. No final, mostre quantas pessoas ainda não atingiram a maioridade e quantas já são maiores (use a biblioteca datetime para comparar com a data atual)

In [ ]:
from datetime import date

anoAtual = date.today().year
print(anoAtual)

qtdMaiorIdade = 0
qtdMenorIdade = 0

for i in range(1, 8):
  anoNasc = int(input(f'Digite o ano de nascimento da {i}ª pessoa: '))
 # print(idade)
  if (anoAtual - anoNasc) >= 18:
    qtdMaiorIdade += 1
  else:
    qtdMenorIdade += 1
print(f'A quantidade de pessoas maiores de idade é: {qtdMaiorIdade}.')
print(f'A quantidade de pessoas menores de idade é: {qtdMenorIdade}.')

2024


KeyboardInterrupt: Interrupted by user

## Exercício Python 55: Faça um programa que leia o peso de cinco pessoas. No final, mostre qual foi o maior e o menor peso lidos.

In [ ]:
pMaior = 0
pMenor = 0

for p in range(1,6):
  peso = float(input(f'Digite o peso (em kg) da {p}ª pessoa: '))
  if p == 1:
    pMaior = peso  # como é a primeira iteração/ocorrência/laço tanto peso maior e peso menor terão o mesmo valor
    pMenor = peso  # como é a primeira iteração tanto peso maior e peso menor terão o mesmo valor
  else:
    if peso >= pMaior:
      pMaior = peso
    elif peso < pMenor: # aqui tanto faz ser um "if" ou "elif"
      pMenor = peso

print(f'O menor peso é: {pMenor} kg. ')
print(f'O maior peso é: {pMaior} kg.')

Digite o peso (em kg) da 1ª pessoa: 90
Digite o peso (em kg) da 2ª pessoa: 120
Digite o peso (em kg) da 3ª pessoa: 45
Digite o peso (em kg) da 4ª pessoa: 69
Digite o peso (em kg) da 5ª pessoa: 75
O menor peso é: 45.0 kg. 
O maior peso é: 120.0 kg.


## Exercício Python 56: Desenvolva um programa que leia o nome, idade e sexo de 4 pessoas. No final do programa, mostre: a média de idade do grupo, qual é o nome do homem mais velho e quantas mulheres têm menos de 20 anos.

In [ ]:
somaIdade = 0
maisVelho = 0
nomeMaisVelho = ''
qtdMenosVinte = 0
media = 0

for pessoas in range(1, 5):
  nome = input(f'Digite o nome da {pessoas}ª pessoa: ').strip().upper()
  idade = int(input(f'Digite a idade da {pessoas}ª pessoa: '))
  sexo = input(f'Digite o sexo (M ou F) da {pessoas}ª pessoa: ').strip().upper()

  # calculando a média de idade de todas as pessoas
  somaIdade += idade
  media = somaIdade / 4

  # definindo o homem mais velho
  if sexo == 'M':
    if pessoas == 0:
      maisVelho = idade
      nomeMaisVelho = nome
    else:
      if idade >= maisVelho:
        maisVelho = idade
        nomeMaisVelho = nome

  # definindo quantas mulheres têm menos de 20 anos
  if sexo == 'F':
    if idade < 20:
      qtdMenosVinte += 1

print('*'*40)
print(f'\nA média de idade de todas as pessoas é: {media} anos.')
print(f'\nO nome do homem mais velho é: {nomeMaisVelho}.')
print(f'\nA quantidade de mulheres com menos de 20 anos é: {qtdMenosVinte}.')


Digite o nome da 1ª pessoa: joao
Digite a idade da 1ª pessoa: 34
Digite o sexo (M ou F) da 1ª pessoa: m
Digite o nome da 2ª pessoa: ana
Digite a idade da 2ª pessoa: 20
Digite o sexo (M ou F) da 2ª pessoa: f
Digite o nome da 3ª pessoa: maria
Digite a idade da 3ª pessoa: 21
Digite o sexo (M ou F) da 3ª pessoa: f
Digite o nome da 4ª pessoa: pedro
Digite a idade da 4ª pessoa: 20
Digite o sexo (M ou F) da 4ª pessoa: m
****************************************

A média de idade de todas as pessoas é: 23.75 anos.

O nome do homem mais velho é: JOAO.

A quantidade de mulheres com menos de 20 anos é: 0.


## Exercício Python 57: Faça um programa que leia o sexo de uma pessoa, mas só aceite os valores ‘M’ ou ‘F’. Caso esteja errado, peça a digitação novamente até ter um valor correto.

In [ ]:
sexo1 = input('Digite o sexo da pessoa [M/F]:')
while sexo1 not in 'MF':
  sexo1 = input('Opção inválida! Responda M ou F. \nDigite o sexo da pessoa [M/F]: ')
print('Muito bem!')

Digite o sexo da pessoa [M/F]:m
Opção inválida! Responda M ou F. 
Digite o sexo da pessoa [M/F]: m
Opção inválida! Responda M ou F. 
Digite o sexo da pessoa [M/F]: M
Muito bem!


## Exercício Python 58: Melhore o jogo do DESAFIO 28 onde o computador vai “pensar” em um número entre 0 e 10. Só que agora o jogador vai tentar adivinhar até acertar, mostrando no final quantos palpites foram necessários para vencer.

In [ ]:
from random import randint

contChances = 0
escolhaPC = randint(1, 10)
escolhaHumano = 0

while escolhaPC != escolhaHumano:
    print
    escolhaHumano = int(input('Digite um número inteiro de 1 a 10: '))
    contChances += 1

print('\nPARABÉNS! Você adivinhou o número que eu estava pensando.')
print(f'Você levou {contChances} tentativas para acertar.')

Digite um número inteiro de 1 a 10: 1
Digite um número inteiro de 1 a 10: 2
Digite um número inteiro de 1 a 10: 3
Digite um número inteiro de 1 a 10: 4

PARABÉNS! Você adivinhou o número que eu estava pensando.
Você levou 4 tentativas para acertar.


## Exercício Python 059: Crie um programa que leia dois valores e mostre um menu na tela:

###[ 1 ] somar
###[ 2 ] multiplicar
###[ 3 ] maior
###[ 4 ] novos números
###[ 5 ] sair do programa

### Seu programa deverá realizar a operação solicitada em cada caso.

In [ ]:
num1 = float(input('Digite o primeiro valor: '))
num2 = float(input('Digite o segundo valor: '))
opcao = 0

while opcao != 5:
    opcao = int(input('Digite a opção para realizar a operação:'
                  '\n[1] - somar'
                  '\n[2] - multiplicar'
                  '\n[3] - maior'
                  '\n[4] - novos números'
                  '\n[5] - sair do programa'
                  '\nOpção desejada: '))

    if opcao == 1:
        soma = num1 + num2
        print(f'A soma de {num1} e {num2} é igual a {soma}.')
    elif opcao == 2:
        mult = num1 * num2
        print(f'A multiplicação entre {num1} e {num2} é igual a {mult}.')
    elif opcao == 3:
        if num1 > num2:
            maior = num1
            print(f'O primeiro valor {num1} é maior que o segundo valor {num2}.')
        elif num2 > num1:
            maior = num2
            print(f'O segundo valor {num2} é maior que o primeiro valor {num1}.')
        else:
            print(f'O primeiro valor {num1} é igual ao segundo valor {num2}.')
    elif opcao == 4:
        num1 = float(input('Digite o primeiro valor: '))
        num2 = float(input('Digite o segundo valor: '))
    else:
        print('Programa finalizado! Até a próxima.')


Digite o primeiro valor: 1
Digite o segundo valor: 2
Digite a opção para realizar a operação:
[1] - somar
[2] - multiplicar
[3] - maior
[4] - novos números
[5] - sair do programa
Opção desejada: 2
A multiplicação entre 1.0 e 2.0 é igual a 2.0.
Digite a opção para realizar a operação:
[1] - somar
[2] - multiplicar
[3] - maior
[4] - novos números
[5] - sair do programa
Opção desejada: 1
A soma de 1.0 e 2.0 é igual a 3.0.
Digite a opção para realizar a operação:
[1] - somar
[2] - multiplicar
[3] - maior
[4] - novos números
[5] - sair do programa
Opção desejada: 3
O segundo valor 2.0 é maior que o primeiro valor 1.0.
Digite a opção para realizar a operação:
[1] - somar
[2] - multiplicar
[3] - maior
[4] - novos números
[5] - sair do programa
Opção desejada: 4
Digite o primeiro valor: 2
Digite o segundo valor: 5
Digite a opção para realizar a operação:
[1] - somar
[2] - multiplicar
[3] - maior
[4] - novos números
[5] - sair do programa
Opção desejada: 2
A multiplicação entre 2.0 e 5.0 é igua

## Exercício Python 060: Faça um programa que leia um número qualquer e mostre o seu fatorial. Exemplo:

### 5! = 5 x 4 x 3 x 2 x 1 = 120

In [ ]:
# minha solução (nada elegante)

num = int(input('Digite um número inteiro para calcular o seu fatorial: '))
print(f'Calculando {num}! {num}x', end='')
mult = 1

while num != 1:
    mult *= num # o acumulador deve ficar antes da subtração do contador
    num -= 1
    if num > 1:
        print(f'{num}', end='x')
print(1, f'= {mult}')


Digite um número inteiro para calcular o seu fatorial: 5
Calculando 5! 5x4x3x2x1 = 120


In [ ]:
# solução do Guanabara

n = int(input('Digite um número para calcular seu fatorial: '))
cont = n
result = 1
while cont > 0:
    print(f'{cont}', end='')
    print(' x ' if cont > 1 else f' = {result}', end='') # ele usa o format, mas eu não gosto kkkkk
    result *= cont # o acumulador deve ficar antes da subtração do contador
    cont -= 1

Digite um número para calcular seu fatorial: 5
5 x 4 x 3 x 2 x 1 = 120

## Exercício Python 61: Refaça o DESAFIO 51, lendo o primeiro termo e a razão de uma PA, mostrando os 10 primeiros termos da progressão usando a estrutura while.

In [ ]:
pTermo = int(input('Digite o primeiro termo da P.A.: '))
razao = int(input('Digite a razão da P.A.: '))
soma = pTermo
c = 1
print(f'1º termo: {pTermo}')
while c < 10: # esse laço será executado apenas 9 vezes, pois queremos 10 termos e já temos 1, o primeiro termo
  soma += razao
  print(f'{c + 1}º termo: {soma}') # {c + 1}, pois eu já tenho o primeiro termo definido
  c += 1

Digite o primeiro termo da P.A.: 3
Digite a razão da P.A.: 3
1º termo: 3
2º termo: 6
3º termo: 9
4º termo: 12
5º termo: 15
6º termo: 18
7º termo: 21
8º termo: 24
9º termo: 27
10º termo: 30


## Exercício Python 62: Melhore o DESAFIO 61, perguntando para o usuário se ele quer mostrar mais alguns termos. O programa encerrará quando ele disser que quer mostrar 0 termos.

In [ ]:
pTermo = int(input('Digite o primeiro termo da P.A.: '))
razao = int(input('Digite a razão da P.A.: '))
soma = pTermo
qtd_termos = 10
c = 1
print(f'1º termo: {pTermo}')
while c < qtd_termos: # esse laço será executado apenas 9 vezes, pois queremos 10 termos e já temos 1, o primeiro termo
  soma += razao
  print(f'{c + 1}º termo: {soma}') # {c + 1}, pois eu já tenho o primeiro termo definido
  c += 1

maisTermos = int(input('Deseja mostrar mais quantos termos?'
                       '\nDigite 0 para sair.'
                       '\nQtd. desejada: '))

if maisTermos == 0:
    print('Programa finalizado!')
else:
    while c < (qtd_termos + maisTermos): # esse laço será executado após a execução do primeiro while, levando em consideração o valor que já consta em "c" devido ao primeiro laço
      soma += razao
      print(f'{c + 1}º termo: {soma}')
      c += 1

Digite o primeiro termo da P.A.: 5
Digite a razão da P.A.: 5
1º termo: 5
2º termo: 10
3º termo: 15
4º termo: 20
5º termo: 25
6º termo: 30
7º termo: 35
8º termo: 40
9º termo: 45
10º termo: 50
Deseja mostrar mais quantos termos?
Digite 0 para sair.
Qtd. desejada: 5
11º termo: 55
12º termo: 60
13º termo: 65
14º termo: 70
15º termo: 75


## Exercício Python 63: Escreva um programa que leia um número N inteiro qualquer e mostre na tela os N primeiros elementos de uma Sequência de Fibonacci. Exemplo:

### 0 – 1 – 1 – 2 – 3 – 5 – 8

In [ ]:
print('*** SEQUÊNCIA DE FIBONACCI ***')
termos = int(input('Digite o números de termos da sequência de Fibonacci: '))
cont = 3        # já mostrei os dois primeiros termos, portanto começa do terceiro
t1 = 0
t2 = 1
prox = 0

print(f'{t1} {t2}', end='')

while cont <= termos:
    t3 = t1 + t2
    print(' {}'. format(t3), end='')
    t1 = t2
    t2 = t3
    cont += 1
print('\n----- FIM -----')

*** SEQUÊNCIA DE FIBONACCI ***
Digite o números de termos da sequência de Fibonacci: 7
0 1 1 2 3 5 8
----- FIM -----


## Exercício Python 64: Crie um programa que leia vários números inteiros pelo teclado. O programa só vai parar quando o usuário digitar o valor 999, que é a condição de parada. No final, mostre quantos números foram digitados e qual foi a soma entre eles (desconsiderando o flag).

In [ ]:
num = 0
soma = 0

while num != 999:
    soma += num
    num = int(input('Digite um número inteiro (digite 999 p/ finalizar o programa): '))

print(f'A soma de todos os valores digitados é: {soma}.')

Digite um número inteiro (digite 999 p/ finalizar o programa): 10
Digite um número inteiro (digite 999 p/ finalizar o programa): 20
Digite um número inteiro (digite 999 p/ finalizar o programa): 60
Digite um número inteiro (digite 999 p/ finalizar o programa): 999
A soma de todos os valores digitados é: 1089.


## Exercício Python 65: Crie um programa que leia vários números inteiros pelo teclado. No final da execução, mostre a média entre todos os valores e qual foi o maior e o menor valores lidos. O programa deve perguntar ao usuário se ele quer ou não continuar a digitar valores.

In [ ]:
# declaração das variáveis

maior = 0
menor = 0
somaNum = 0
resp = ''
contNum = 0
num = 0

# utilizando o laço while para resolver o problema

while resp != 'N':
    num = int(input('Digite um número inteiro: '))
    resp = input('\nDeseja digitar mais números? [Sim/Não]'
                 '\nOpção desejada: ').strip().upper()[0] # [0] serve para selecionar apenas a primeira letra da string "Sim" ou "Não", assim fica mais intuitivo
    while resp not in 'SN': # como eu indiquei apenas o primeiro caractere [0] da string, não há problema desse teste lógico ser feito dessa maneira
        print('\nOpção inválida!')
        resp = input('\nDeseja digitar mais números? [Sim/Não]'
                     '\nOpção desejada: ').strip().upper()[0]

    # calculando a soma de todos os números digitados
    somaNum += num
    contNum += 1

    # definindo o maior e o menor número
    if contNum == 1:
        maior = num
        menor = num
    elif num > maior:
        maior = num
    elif num < menor:
        menor = num

# calculando a média dos números digitados
media = somaNum / contNum

# exibição dos resultados

print(f'A média do(s) {contNum} número(s) digitado(s) é de: {media:.2f}.')
print(f'O maior número digitado dentre o(s) {contNum} número(s) digitado(s) é de: {maior}.')
print(f'O menor número digitado dentre o(s) {contNum} número(s) digitado(s) é de: {menor}.')

Digite um número inteiro: 5

Deseja digitar mais números? [Sim/Não]
Opção desejada: s
Digite um número inteiro: 6

Deseja digitar mais números? [Sim/Não]
Opção desejada: n
A média do(s) 2 número(s) digitado(s) é de: 5.50.
O maior número digitado dentre o(s) 2 número(s) digitado(s) é de: 6.
O menor número digitado dentre o(s) 2 número(s) digitado(s) é de: 5.


## Exercício Python 66: Crie um programa que leia números inteiros pelo teclado. O programa só vai parar quando o usuário digitar o valor 999, que é a condição de parada. No final, mostre quantos números foram digitados e qual foi a soma entre elas (desconsiderando o flag).

In [ ]:
# declaração das variáveis
num = 0
soma = 0
qtdNum = 0

# resolução do problema

while True:
    num = int(input('Digite um número inteiro: '))
    if num == 999:
        break
    soma += num
    qtdNum += 1

print(f'A soma dos números digitados é: {soma}.')
print(f'A quantidade de números digitados é: {qtdNum}.')

# exibição dos resultados

Digite um número inteiro: 5
Digite um número inteiro: 7
Digite um número inteiro: 3
Digite um número inteiro: 8
Digite um número inteiro: 57
Digite um número inteiro: 46
Digite um número inteiro: 999
A soma dos números digitados é: 126.
A quantidade de números digitados é: 6.


## Exercício Python 67: Faça um programa que mostre a tabuada de vários números, um de cada vez, para cada valor digitado pelo usuário. O programa será interrompido quando o número solicitado for negativo.

In [ ]:
# declaração das variáveis
num = 0

# resolução do problema
while True:
    num = int(input('\nDigite um número para saber sua tabuada: '))
    if num < 0:
        print('Me desculpe mas não posso calcular a tabuada de um número negativo!')
        break
    for c in range(1, 11):
        print(f'{num} x {c} = {num*c}')


Digite um número para saber sua tabuada: 3
3 x 1 = 3
3 x 2 = 6
3 x 3 = 9
3 x 4 = 12
3 x 5 = 15
3 x 6 = 18
3 x 7 = 21
3 x 8 = 24
3 x 9 = 27
3 x 10 = 30

Digite um número para saber sua tabuada: 5
5 x 1 = 5
5 x 2 = 10
5 x 3 = 15
5 x 4 = 20
5 x 5 = 25
5 x 6 = 30
5 x 7 = 35
5 x 8 = 40
5 x 9 = 45
5 x 10 = 50

Digite um número para saber sua tabuada: 7
7 x 1 = 7
7 x 2 = 14
7 x 3 = 21
7 x 4 = 28
7 x 5 = 35
7 x 6 = 42
7 x 7 = 49
7 x 8 = 56
7 x 9 = 63
7 x 10 = 70

Digite um número para saber sua tabuada: 2
2 x 1 = 2
2 x 2 = 4
2 x 3 = 6
2 x 4 = 8
2 x 5 = 10
2 x 6 = 12
2 x 7 = 14
2 x 8 = 16
2 x 9 = 18
2 x 10 = 20

Digite um número para saber sua tabuada: -2
Me desculpe mas não posso calcular a tabuada de um número negativo!


## Exercício Python 68: Faça um programa que jogue par ou ímpar com o computador. O jogo só será interrompido quando o jogador perder, mostrando o total de vitórias consecutivas que ele conquistou no final do jogo.

In [ ]:
# importando as bibliotecas necessárias
from random import randint
# definição das variáveis
contVit = 0
escolhaPCNum = randint(1, 10)
escolhaPC = ''
escolhaHumano = ''
escolhaHumanoNum = 0
somaNum = 0
contVit = 0


# definição de quem pede par e quem pede ímpar

while True:
    # entrada de dados

    while escolhaHumano not in ('PAR', 'IMPAR'):
        escolhaHumano = input('Você quer PAR ou IMPAR?'
                              '\nOpção desejada: ').strip().upper()

    # definindo se o PC escolherá PAR ou ÍMPAR a depender da escolha do humano
    if escolhaHumano == 'PAR':
        print('PC: Então eu quero IMPAR!')
        escolhaPC = 'IMPAR'
    elif escolhaHumano == 'IMPAR':
        print('PC: Então eu quero PAR!')
        escolhaPC = 'PAR'

    escolhaHumanoNum = int(input('Digite um número de 1 a 10: '))

    somaNum = escolhaPCNum + escolhaHumanoNum

    if somaNum % 2 == 0 and escolhaHumano == 'PAR':
        print('Dessa vez você me venceu, humano!')
        contVit += 1
    else:
        print('Eu te venci, humano! Muahahahaha!')
        break
    if somaNum % 2 == 1 and escolhaHumano == 'IMPAR':
        print('Dessa vez você me venceu, humano!')
        contVit += 1
    else:
        print('Eu te venci, humano! Muahahahaha!')
        break
print(f'Você me venceu {contVit} vez(es).')

# saída de dados

print(f'Escolha humano: {escolhaHumano}'
      f'\nNúmero humano: {escolhaHumanoNum}'
      f'\nEscolha PC: {escolhaPC}'
      f'\nNúmero PC: {escolhaPCNum}')

# erros corrigidos pelo chatGPT:

"""Destaque das Diferenças:

ERRO NAS CONDIÇÕES: No seu código original, as duas verificações (if) estavam independentes e podiam ambas ser executadas, o que não era desejado. Na versão corrigida, usei elif para garantir que apenas uma das condições seja avaliada e executada.
Além disso, a condição correta foi movida para a segunda verificação, e agora ela verifica se somaNum % 2 == 1 quando escolhaHumano == 'IMPAR'.

    # condições corrigidas
    if somaNum % 2 == 0 and escolhaHumano == 'PAR':
        print('Dessa vez você me venceu, humano!')
        contVit += 1
    elif somaNum % 2 == 1 and escolhaHumano == 'IMPAR':
        print('Dessa vez você me venceu, humano!')
        contVit += 1
    else:
        print('Eu te venci, humano! Muahahahaha!')
        break

ERRO AO NÃO REINICIAR AS VARIÁVEIS:
    # resetando escolhas para a próxima rodada
    escolhaHumano = ''
    escolhaPCNum = randint(1, 10)

NO CÓDIGO ABAIXO CONSTAM AS CORREÇÕES, DEIXEI O CÓDIGO ORIGINALMENTE ERRADO, PARA VERIFICAÇÕES NO FUTURO, AFIM DE EVITAR NOVOS ERROS E TAMBÉM COMO FONTE DE OBSERVAÇÃO DO AVANÇO DE APRENDIZADO..."""

Você quer PAR ou IMPAR?
Opção desejada: par
PC: Então eu quero IMPAR!
Digite um número de 1 a 10: 3
Dessa vez você me venceu, humano!
Eu te venci, humano! Muahahahaha!
Você me venceu 1 vez(es).
Escolha humano: PAR
Número humano: 3
Escolha PC: IMPAR
Número PC: 9


In [ ]:
# código feito por mim, porém, com o erro nas condições IF-ELSE corrigido pelo chatGPT

from random import randint

# definição das variáveis
contVit = 0
escolhaPCNum = randint(1, 10)
escolhaPC = ''
escolhaHumano = ''
escolhaHumanoNum = 0
somaNum = 0

while True:
    # entrada de dados
    while escolhaHumano not in ('PAR', 'IMPAR'):
        escolhaHumano = input('Você quer PAR ou IMPAR?\nOpção desejada: ').strip().upper()

    # definindo se o PC escolherá PAR ou ÍMPAR a depender da escolha do humano
    if escolhaHumano == 'PAR':
        print('PC: Então eu quero IMPAR!')
        escolhaPC = 'IMPAR'
    elif escolhaHumano == 'IMPAR':
        print('PC: Então eu quero PAR!')
        escolhaPC = 'PAR'

    escolhaHumanoNum = int(input('Digite um número de 1 a 10: '))

    somaNum = escolhaPCNum + escolhaHumanoNum

    print(f'Número PC: {escolhaPCNum}')
    print(f'Número humano: {escolhaHumanoNum}')
    print(f'Soma dos números: {somaNum}')

    # condições corrigidas
    if somaNum % 2 == 0 and escolhaHumano == 'PAR':
        print('Dessa vez você me venceu, humano!')
        contVit += 1
    elif somaNum % 2 == 1 and escolhaHumano == 'IMPAR':
        print('Dessa vez você me venceu, humano!')
        contVit += 1
    else:
        print('Eu te venci, humano! Muahahahaha!')
        break

    # resetando escolhas para a próxima rodada
    escolhaHumano = ''
    escolhaPCNum = randint(1, 10)

print(f'Você me venceu {contVit} vez(es).')


Você quer PAR ou IMPAR?
Opção desejada: par
PC: Então eu quero IMPAR!
Digite um número de 1 a 10: 3
Número PC: 2
Número humano: 3
Soma dos números: 5
Eu te venci, humano! Muahahahaha!
Você me venceu 0 vez(es).


## Exercício Python 69: Crie um programa que leia a idade e o sexo de várias pessoas. A cada pessoa cadastrada, o programa deverá perguntar se o usuário quer ou não continuar. No final, mostre:

### A) quantas pessoas tem mais de 18 anos.
### B) quantos homens foram cadastrados.
### C) quantas mulheres tem menos de 20 anos.

In [ ]:
# inicializando as variáveis

idade = 0
sexo = ''
contIdade = 0
contHomens = 0
mulherMenosDeVinte = 0
resp = ''

# resolução do problema
while True:

    # entrada de dados

    idade = int(input('Digite a idade da pessoa a ser cadastrada: '))

    sexo = (input('Digite o sexo da pessoa a ser cadastrada [M/F]: ')).strip().upper()
    while sexo not in 'MF':
        sexo = (input('Digite o sexo da pessoa a ser cadastrada [M/F]: ')).strip().upper()

    # resolução das condiçoes
    if idade < 18:
        contIdade += 1

    if sexo == 'M':
        contHomens += 1

    if idade < 20 and sexo == 'F':
        mulherMenosDeVinte += 1

    resp = input('\nDeseja continuar com os cadastros? [S/N]'
                     '\nOpção selecionada: ').strip().upper()
    while resp not in 'SN':
        resp = input('\nDeseja continuar com os cadastros? [S/N]'
                     '\nOpção selecionada: ').strip().upper()

    if resp == 'N':
        break

# apresentação dos resultados
print(f'Você cadastrou {contIdade} pessoa(s) menor(es) de 18 anos.')
print(f'Você cadastrou {contHomens} homem(ns).')
print(f'Você cadastrou {mulherMenosDeVinte} mulher(es) menor(es) de 20 anos.')

Digite a idade da pessoa a ser cadastrada: 17
Digite o sexo da pessoa a ser cadastrada [M/F]: f

Deseja continuar com os cadastros? [S/N]
Opção selecionada: s
Digite a idade da pessoa a ser cadastrada: 20
Digite o sexo da pessoa a ser cadastrada [M/F]: m

Deseja continuar com os cadastros? [S/N]
Opção selecionada: s
Digite a idade da pessoa a ser cadastrada: 23
Digite o sexo da pessoa a ser cadastrada [M/F]: f

Deseja continuar com os cadastros? [S/N]
Opção selecionada: s
Digite a idade da pessoa a ser cadastrada: 14
Digite o sexo da pessoa a ser cadastrada [M/F]: m

Deseja continuar com os cadastros? [S/N]
Opção selecionada: n
Você cadastrou 2 pessoa(s) menor(es) de 18 anos.
Você cadastrou 2 homem(ns).
Você cadastrou 1 mulher(es) menor(es) de 20 anos.


## Exercício Python 70: Crie um programa que leia o nome e o preço de vários produtos. O programa deverá perguntar se o usuário vai continuar ou não. No final, mostre:

### A) qual é o total gasto na compra.
### B) quantos produtos custam mais de R$1000.
### C) qual é o nome do produto mais barato.

In [ ]:
# inicializando as variáveis
resp = ''
precoProd = 0
nomeProd = ''
totGasto = 0
contMil = 0
contProdutos = 0
nomeProdMaisBarato = ''


# resolução do problema
while True:

    # entrada de dados
    nomeProd = input('Digite o nome do produto: ').strip().upper()
    precoProd = float(input('Digite o valor do produto: '))

    totGasto += precoProd
    contProdutos += 1

    # condições
    if precoProd > 1000:
        contMil += 1

    if contProdutos == 1:
        prodMaisBarato = precoProd
        nomeProdMaisBarato = nomeProd
    elif precoProd < prodMaisBarato:
        prodMaisBarato = precoProd
        nomeProdMaisBarato = nomeProd

    resp = input('\nDeseja continuar com os cadastros de produtos? [S/N]'
                     '\nOpção selecionada: ').strip().upper()
    while resp not in 'SN':
        resp = input('\nDeseja continuar com os cadastros de produtos? [S/N]'
                     '\nOpção selecionada: ').strip().upper()
    if resp == 'N':
        break


print(f'\nQuantidade de produto(s) comprado(s): {contProdutos}.'
      f'O valor total da compra é de R$ {totGasto}.')
print(f'\nO total de produtos que custam mais de R$ 1.000,00 são: {contMil} produtos.')
print(f'\nO produto mais barato é o {nomeProdMaisBarato} no valor de R$ {prodMaisBarato}.')

Digite o nome do produto: arroz
Digite o valor do produto: 10.90

Deseja continuar com os cadastros de produtos? [S/N]
Opção selecionada: s
Digite o nome do produto: feijão
Digite o valor do produto: 5.60

Deseja continuar com os cadastros de produtos? [S/N]
Opção selecionada: s
Digite o nome do produto: tv led
Digite o valor do produto: 1200

Deseja continuar com os cadastros de produtos? [S/N]
Opção selecionada: s
Digite o nome do produto: ps5
Digite o valor do produto: 4799.99

Deseja continuar com os cadastros de produtos? [S/N]
Opção selecionada: s
Digite o nome do produto: rtx 4070 ti
Digite o valor do produto: 7499

Deseja continuar com os cadastros de produtos? [S/N]
Opção selecionada: n
O total gasto na compra é de R$ 13515.49.
O total de produtos que custam mais de R$ 1.000,00 são: 3 produtos.
O produto mais barato é o FEIJÃO no valor de R$ 5.6.


## Exercício Python 071: Crie um programa que simule o funcionamento de um caixa eletrônico. No início, pergunte ao usuário qual será o valor a ser sacado (número inteiro) e o programa vai informar quantas cédulas de cada valor serão entregues. OBS:

### considere que o caixa possui cédulas de 50 reais, 20 reais, 10 reais e 1 real.

In [ ]:
tupla = ('Carro', 2.33, 43)
print(tupla[0])

Carro


In [ ]:
while True:
    valor = int(input('Qual o valor do saque (apenas valores inteiros): '))

    # 235
    notas50 = valor // 50 # dado o valor, a divisão inteira resultará em uma quantidade de cédulas de 50 reais
    resto = valor % 50 # retornará o resto da divisão do valor por 50

    notas20 = resto // 20 # dado o valor, a divisão inteira resultará em uma quantidade de cédulas de 20 reais
    resto1 = resto % 20 # retornará o resto da divisão do valor por 20

    notas10 = resto1 // 10 # dado o valor, a divisão inteira resultará em uma quantidade de cédulas de 10 reais
    resto2 = resto1 % 10 # retornará o resto da divisão do valor por 10

    notas5 = resto2 // 1 # dado o valor, a divisão inteira resultará em uma quantidade de cédulas de 5 reais

    # apresentando o resultados das quantidades de cédulas de 50, 20, 10 e 5 reais

    print(f'\nVocê sacou R$ {valor}.'
      f'\nReceberá {notas50} nota(s) de R$ 50,00'
      f'\nReceberá {notas20} nota(s) de R$ 20,00'
      f'\nReceberá {notas10} nota(s) de R$ 10,00'
      f'\nReceberá {notas5} nota(s) de R$ 1,00')

    # perguntando ao usuário se ele deseja continuar com a execução do programa

    resp = input('\nDeseja sacar mais algum valor? [S/N]'
                     '\nOpção selecionada: ').strip().upper()
    while resp not in 'SN':
        resp = input('\nDeseja sacar mais algum valor? [S/N]'
                     '\nOpção selecionada: ').strip().upper()
    if resp == 'N':
        break


Qual o valor do saque (apenas valores inteiros): 50

Você sacou R$ 50.
Receberá 1 nota(s) de R$ 50,00
Receberá 0 nota(s) de R$ 20,00
Receberá 0 nota(s) de R$ 10,00
Receberá 0 nota(s) de R$ 1,00

Deseja sacar mais algum valor? [S/N]
Opção selecionada: s
Qual o valor do saque (apenas valores inteiros): 40

Você sacou R$ 40.
Receberá 0 nota(s) de R$ 50,00
Receberá 2 nota(s) de R$ 20,00
Receberá 0 nota(s) de R$ 10,00
Receberá 0 nota(s) de R$ 1,00

Deseja sacar mais algum valor? [S/N]
Opção selecionada: s
Qual o valor do saque (apenas valores inteiros): 65

Você sacou R$ 65.
Receberá 1 nota(s) de R$ 50,00
Receberá 0 nota(s) de R$ 20,00
Receberá 1 nota(s) de R$ 10,00
Receberá 5 nota(s) de R$ 1,00

Deseja sacar mais algum valor? [S/N]
Opção selecionada: n
